library/core/src/escape.rs

```diff
@@ -0,0 +1,97 @@
+//! Helper code for character escaping.
```

Comment:
The reason I didn’t want to put it inside `ascii` is because of slight differences in escaping. `'\0'.escape_default()` produces `\u{0}` while `0u8.escape_default()` produces `\x00` (also `'\0'.escape_debug()` produces `\0`).  The way I was thinking about it is that `std::ascii::escape_default` and `char::escape_default` both use `EscapeIterInner` so they are at the same ‘hierarchy’ meaning that `EscapeIterInner` shouldn’t be inside `std::ascii`.

But yes, I understand your concerns and I’m not completely convinced either.  Maybe I’m just overthinking this?